source/parser.hera

```diff
@@ -1870,18 +1869,22 @@ BinaryOpSymbol
       token: "in",
       special: true,
     }
-  CoffeeOfEnabled "in" NonIdContinue ->
+  ( "is" NonIdContinue __ "in" NonIdContinue ) / ( CoffeeOfEnabled "in" NonIdContinue ) ->
     return {
       ref: module.getRef("indexOf"),
       suffix: " >= 0",
       special: true,
     }
-  CoffeeOfEnabled "not" NonIdContinue __ "in" NonIdContinue ->
+  ( "is" NonIdContinue __ "not" NonIdContinue __ "in" NonIdContinue ) / ( CoffeeOfEnabled "not" NonIdContinue __ "in" NonIdContinue ) ->
     return {
       ref: module.getRef("indexOf"),
       suffix: " < 0",
       special: true,
     }
+  # NOTE: "is not" must come after "is not in"
+  "is" NonIdContinue __ "not" NonIdContinue -> "!=="
```

Comment:
We probably want to use `_*` here because if the following line starts with an indented `not` it will get eaten up by `__ "not"`.

Reply:
Isn't that intentional? What would `is` at the end of a line mean? I was just matching the existing style from CoffeeScript's `not in` above. See also https://coffeescript.org/#try:5%20not%0A%0Ain%2010

Reply:
This does raise an ambiguity in `coffeeCompat` mode: `x is not y` already means `x !== not y`. This PR changes that behavior to `x !== y`... 🙁

Reply:
> Isn't that intentional? What would `is` at the end of a line mean? I was just matching the existing style from CoffeeScript's `not in` above. See also https://coffeescript.org/#try:5%20not%0A%0Ain%2010

That's probably correct. The difficulty of whitespace sensitivity is you always have to consider it when writing the parser :)